containers/Filters/AddFilterModal.js

```diff
@@ -0,0 +1,196 @@
+import React, { useState } from 'react';
+import PropTypes from 'prop-types';
+import { c } from 'ttag';
+import {
+    Button,
+    FormModal,
+    ResetButton,
+    PrimaryButton,
+    useEventManager,
+    useNotifications,
+    useApiWithoutResult
+} from 'react-components';
+import { newFilter, format as formatFilter } from 'proton-shared/lib/filters/factory';
+import { validate, validateComplex } from 'proton-shared/lib/filters/validator';
+import { addTreeFilter, updateFilter } from 'proton-shared/lib/api/filters';
+import { noop } from 'proton-shared/lib/helpers/function';
+
+import ConditionsEditor from '../../components/Filters/editor/Conditions';
+import ActionsEditor from '../../components/Filters/editor/Actions';
+import OperatorEditor from '../../components/Filters/editor/Operator';
+import SieveEditor from '../../components/Filters/editor/Sieve';
+import PreviewFilter from '../../components/Filters/editor/Preview';
+import NameEditor from '../../components/Filters/editor/Name';
+
+import './AddFilterModal.css';
+
+function AddFilterModal({ filter, type, mode, onEdit, ...props }) {
```

Comment:
For me it makes more sense to have `mode` for `simple` or `complex`. And `method` for `create` or `update`.

Reply:
method too technical imho. Mode is a behavior.